### PR TITLE
[New Python API] Add new python API based on pybind11 

### PR DESCRIPTION
**Pull request by akuporos**
This is POC branch where Python API is based on pybind11

Examples of using new Python API -- https://github.com/akuporos/openvino/pull/45/files

ticket #33021

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
> Examples of using new Python API -- https://github.com/akuporos/openvino/pull/45/files

I don't see example here :)

File: inference-engine/ie_bridges/python/src/openvino/inference_engine/CMakeLists.txt [hunk 1]
```diff
@@ -60,13 +60,17 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     target_compile_options(${TARGET_NAME} PRIVATE "-Wno-error=register")
 endif()
 
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
```
Comment (ilya-lavrenov) [inline inference-engine/ie_bridges/python/src/openvino/inference_engine/CMakeLists.txt:63]: don't use this package. It does not support `-DPYTHON_EXECUTABLE`

Comment (akuporos) [reply inference-engine/ie_bridges/python/src/openvino/inference_engine/CMakeLists.txt]: done

File: runtime/bindings/python/src/pyopenvino/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+project (pyopenvino)
+
+if(DEFINED OpenVINO_SOURCE_DIR)
+    set(InferenceEngine_LIBRARIES inference_engine)
+else()
+    find_package(InferenceEngine REQUIRED)
+    set(InferenceEngine_LIBRARIES IE::inference_engine)
+endif()
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/src/pyopenvino/CMakeLists.txt:11]: I believe current python is based on new OpenVINOConfig.cmake. Can we use it?

File: runtime/bindings/python/tests/test_inference_engine/test_infer_request.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+import numpy as np
+import os
+import pytest
+
+from openvino import Core, Blob, TensorDesc, StatusCode
+
+
+def image_path():
+    path_to_repo = os.environ["DATA_PATH"]
+    path_to_img = os.path.join(path_to_repo, "validation_set", "224x224", "dog.bmp")
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/tests/test_inference_engine/test_infer_request.py:13]: why do we need image? Tests are work on dummy data

File: runtime/bindings/python/tests/test_inference_engine/test_infer_request.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+def model_path(is_myriad=False):
+    path_to_repo = os.environ["MODELS_PATH"]
+    if not is_myriad:
+        test_xml = os.path.join(path_to_repo, "models", "test_model", "test_model_fp32.xml")
+        test_bin = os.path.join(path_to_repo, "models", "test_model", "test_model_fp32.bin")
+    else:
+        test_xml = os.path.join(path_to_repo, "models", "test_model", "test_model_fp16.xml")
+        test_bin = os.path.join(path_to_repo, "models", "test_model", "test_model_fp16.bin")
+    return (test_xml, test_bin)
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/tests/test_inference_engine/test_infer_request.py:25]: can we generate IRs at the beginning of the tests and remove upon the finish? It's not convenient to work with env variables and extra repositories

File: runtime/bindings/python/src/pyopenvino/CMakeLists.txt (hunk 2 above)
Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/CMakeLists.txt:8]: done

File: runtime/bindings/python/tests/test_inference_engine/test_infer_request.py (hunk 3 above)
Comment (akuporos) [reply runtime/bindings/python/tests/test_inference_engine/test_infer_request.py:13]: discussed offline. Created ticket: 67090

File: runtime/bindings/python/tests/test_inference_engine/test_infer_request.py (hunk 4 above)
Comment (akuporos) [reply runtime/bindings/python/tests/test_inference_engine/test_infer_request.py:25]: discussed offline. Created ticket: 67090

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**